### ISSUE TITLE
[BUG] Protectable Containers - List api doesn't work

### ISSUE THREAD

**Issue by Dhoorban** [Azure/azure-sdk-for-java#35817]
**Describe the bug**
I tried to list protectable containers both by using java sdk and using rest api. Both don't work. 
Here's the Java sample link : 
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/recoveryservicesbackup/azure-resourcemanager-recoveryservicesbackup/SAMPLE.md#protectablecontainers_list 

And here's the Rest api's link:
https://learn.microsoft.com/en-us/rest/api/backup/protectable-containers/list

***Exception or Stack Trace***
In both cases i'm getting this error message: 
```
{
  "error": {
    "code": "BMSUserErrorInvalidInput",
    "message": "The user input given for the call is invalid. Please check the required inputs"
  }
}
```

**To Reproduce**
You can run the sample code provided by microsoft themselves to reproduce the issue. 
Rest: https://learn.microsoft.com/en-us/rest/api/backup/protectable-containers/list
Java:  https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/recoveryservicesbackup/azure-resourcemanager-recoveryservicesbackup/SAMPLE.md#protectablecontainers_list 

**Expected behavior**
Expecting a list of protectable VMs in json format. 

Screenshots:
![image](https://github.com/Azure/azure-sdk-for-java/assets/4924392/c793dc29-0404-4b3f-885e-6c689df4a57d)
![image](https://github.com/Azure/azure-sdk-for-java/assets/4924392/a604ded0-0040-445d-bcfd-9a48ae9d77c0)

**Comment by joshfree** [comment]
@weidongxu-microsoft could you please assist @Dhoorban with this github issue?  Thank you

**Comment by weidongxu-microsoft** [comment]
@Dhoorban

I assume you have these pre-requisite resources (resource group, recovery services vault, fabric, they all exist in your subscription), correct?

I am routing this to backend. If these resources are available, and api-version provided, I don't think there be any other requirements on input.

**Comment by Dhoorban** [comment]
> @Dhoorban
> 
> I assume you have these pre-requisite resources (resource group, recovery services vault, fabric, they all exist in your subscription), correct?
> 
> I am routing this to backend. If these resources are available, and api-version provided, I don't think there be any other requirements on input.

Yes they exist in my subscription. If they don't exist, then i will get a different error ie ResourceGroupNotFound . For the fabric parameter i used Azure as the value.

**Comment by weidongxu-microsoft** [comment]
@Dhoorban

Thanks for the confirmation. Did you tried not using the `$filter` query parameter?

The line in REST page seems not correct (it is not percentage encoded).
But the one to SDK would be automatically encoded by SDK before sending to backend.

**Comment by Dhoorban** [comment]
> @Dhoorban
> 
> Thanks for the confirmation. Did you tried not using the `$filter` query parameter?
> 
> The line in REST page seems not correct (it is not percentage encoded). But the one to SDK would be automatically encoded by SDK before sending to backend.

I tried that too . I'm getting the same error.
![image](https://github.com/Azure/azure-sdk-for-java/assets/4924392/fc3a45de-de3f-4e11-b718-63be96a72c47)

**Comment by v-hongli1** [comment]
@weidongxu-microsoft 
Even without using `$filter`, I'm still getting the same error.
![image](https://github.com/Azure/azure-sdk-for-java/assets/74638143/e7ec0576-7cf4-459e-89ba-a897c38d4470)
So I think it is more appropriate for this issue to be handled by backend.

**Comment by weidongxu-microsoft** [comment]
Thanks Hong.

Agree that the bug is very likely in backend.
